x/cash/dynamicfee_test.go

```
@@ -0,0 +1,240 @@
+		wantDeliverErr   error
+		wantDeliverTxFee x.Coin
+		wantGasPayment   int64
+	}{
+		"on success full transaction fee is charged": {
+			signers: []weave.Condition{perm1},
+			handler: &handlerMock{},
+			initWallets: []orm.Object{
+				walletObj(perm1.Address(), 1, 0, "BTC"),
+			},
+			minumumFee:       x.NewCoin(0, 23, "BTC"),
+			txFee:            x.NewCoin(0, 421, "BTC"),
+			wantCheckTxFee:   x.NewCoin(0, 421, "BTC"),
+			wantDeliverTxFee: x.NewCoin(0, 842, "BTC"),
+			wantGasPayment:   421,
+		},
+		"on a handler check failure minumum fee is charged": {
+			signers: []weave.Condition{perm1},
+			handler: &handlerMock{checkErr: ErrTestingError},
+			initWallets: []orm.Object{
+				walletObj(perm1.Address(), 1, 0, "BTC"),
+			},
+			minumumFee:     x.NewCoin(0, 23, "BTC"),
+			txFee:          x.NewCoin(0, 421, "BTC"),
+			wantCheckErr:   ErrTestingError,
+			wantCheckTxFee: x.NewCoin(0, 23, "BTC"),
+		},
+		"on inssuficient fee funds minimum fee is charged": {
+			signers: []weave.Condition{perm1},
+			initWallets: []orm.Object{
+				walletObj(perm1.Address(), 0, 100, "BTC"),
+			},
+			minumumFee:     x.NewCoin(0, 23, "BTC"),
+			txFee:          x.NewCoin(0, 421, "BTC"), // Wallet has not enough.
+			wantCheckErr:   errors.ErrInsufficientAmount,
+			wantCheckTxFee: x.NewCoin(0, 23, "BTC"),
+		},
+		"on inssuficient funds minumum fee withdraw fails": {
+			signers: []weave.Condition{perm1},
+			initWallets: []orm.Object{
+				walletObj(perm1.Address(), 0, 1, "BTC"),
+			},
+			minumumFee:     x.NewCoin(0, 23, "BTC"),  // Wallet has not enough.
+			txFee:          x.NewCoin(0, 421, "BTC"), // Wallet has not enough.
+			wantCheckErr:   errors.ErrInsufficientAmount,
+			wantCheckTxFee: x.Coin{},
+		},
+		"on transaction fee ticker missmatch minimum fee with no currency accepts anything": {
+			signers: []weave.Condition{perm1},
+			initWallets: []orm.Object{
+				walletObj(perm1.Address(), 1, 0, "BTC"),
+			},
+			minumumFee:     x.NewCoin(0, 23, ""),
```

As mentioned above, I consider this minimumFee illegal and we don't need to handle it. It should return an error when parsing (and when running)

---

Adressed in https://github.com/iov-one/weave/pull/364/commits/5bdc3babe4b936726d84774564b7758710125673